functional-tests/src/main/scala/com/vertica/spark/functests/EndToEndTests.scala

```
@@ -638,6 +647,9 @@ class EndToEndTests(readOpts: Map[String, String], writeOpts: Map[String, String
     assert(r2 == (n + 1))
 
     stmt.execute("drop table " + tableName1)
+    fsLayer.removeDir(fsConfig.address)
+    // Need to recreate the root directory for the afterEach assertion check
+    fsLayer.createDir(fsConfig.address, "777")
```

Why directory should be recreated? If it is absolutely required suggesting to extract it in a separate function to avoid copy pasting  in all tests.

---

The directory needs to be recreated because that command removes the root directory. So when our afterEach logic checks if our directory has files in it, it will fail on path does not exist instead of checking if it's empty. I believe that is only used in 7 test cases, but I can still extract it in a function if that's preferred.